extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/utils.js

```diff
@@ -404,12 +421,132 @@ var Utils = {
    * @returns {Object} Wrapper object for sanitized markup.
    */
   createMarkup: function(html) {
     return {__html: html};
   },
 
+  /**
+   * Deep merge arrays and array values
+   *
+   * @function arrayDeepMerge
+   * @param {Array} target - An array that will receive new items if additional items are passed
+   * @param {Array} source - An array containing additional items to merge into target
+   * @param {Object} optionsArgument - parameters passed to parent DeepMerge function, i.e. -
+   *        arrayMerge - https://github.com/KyleAMathews/deepmerge#arraymerge
+   *        clone - https://github.com/KyleAMathews/deepmerge#clone
+   *        arrayUnionBy - key used to compare Objects being merged, i.e. button name
+   *        arrayFusion - method used to merge arrays ['replace', 'deepmerge']
+   *        buttonArrayFusion - method used to merge button array ['replace', 'prepend', 'deepmerge']
+   *        arraySwap - swaps target/source
+   * @returns {Array} new merged array with items from both target and source
+   */
+  arrayDeepMerge: function(target, source, optionsArgument) {
+    if (source && source.length) {
+      // if source is button and buttonArrayFusion is 'replace', return source w/o merge
+      if (source[0][optionsArgument.arrayUnionBy] && optionsArgument.buttonArrayFusion === 'replace') {
+        return source;
+      }
+      // if source is not button and arrayFusion is 'replace', return source w/o merge
+      else if (!source[0][optionsArgument.arrayUnionBy] && optionsArgument.arrayFusion !== 'deepmerge') {
+        return source;
+      }
+    }
+
+    var targetArray = optionsArgument.arraySwap ? source : target;
+    var sourceArray = optionsArgument.arraySwap ? target : source;
+    var self = this;
+    var uniqueSourceArray = sourceArray.slice(); //array used to keep track of objects that do not exist in target
+    var destination = targetArray.slice();
+
+    sourceArray.forEach(function(sourceItem, i) {
+      if (typeof destination[i] === 'undefined') {
+        destination[i] = self._cloneIfNecessary(sourceItem, optionsArgument);
+      }
+      else if (self._isMergeableObject(sourceItem)) {
+        // custom merge for buttons array, used to maintain source sort order
+        if (sourceItem[optionsArgument.arrayUnionBy]) {
+          targetArray.forEach(function(targetItem, j) {
+            // gracefully merge buttons by name
+            if (sourceItem[optionsArgument.arrayUnionBy] === targetItem[optionsArgument.arrayUnionBy]) {
+              var targetObject = optionsArgument.arraySwap ? sourceItem : targetItem;
+              var sourceObject = optionsArgument.arraySwap ? targetItem : sourceItem;
+              destination[j] = DeepMerge(targetObject, sourceObject, optionsArgument);
+
+              // prunes uniqueSourceArray to unique items not in target
+              if (optionsArgument.buttonArrayFusion === 'prepend' && uniqueSourceArray && uniqueSourceArray.length) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)